packages/kbn-expandable-flyout/src/components/index.tsx

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React, { useCallback, useMemo } from 'react';
+import { css } from '@emotion/react';
+import type { EuiFlyoutProps } from '@elastic/eui';
+import { EuiFlexGroup, EuiFlyout } from '@elastic/eui';
+import { useExpandableFlyoutContext } from './context';
+import { PreviewSection } from './preview_section';
+import { RightSection } from './right_section';
+import type { FlyoutPanel, Panel } from '../models/panel';
+import { LeftSection } from './left_section';
+
+export interface ExpandableFlyoutProps extends EuiFlyoutProps {
+  /**
+   * List of all registered panels available for render
+   */
+  registeredPanels: Panel[];
+  /**
+   * Propagate out EuiFlyout onClose event
+   */
+  handleOnFlyoutClosed?: () => void;
+}
+
+/**
+ * Expandable flyout UI React component.
+ * Displays 3 sections (right, left, preview) depending on the panels in the context.
+ */
+export const ExpandableFlyout: React.FC<ExpandableFlyoutProps> = ({
+  registeredPanels,
+  handleOnFlyoutClosed,
+  ...flyoutProps
+}) => {
+  const {
+    panels: { left, right, preview },
+    closeFlyout,
+  } = useExpandableFlyoutContext();
+
+  const onClose = useCallback(() => {
+    if (handleOnFlyoutClosed) handleOnFlyoutClosed();
+    closeFlyout();
+  }, [closeFlyout, handleOnFlyoutClosed]);
+
+  const leftSection = useMemo(
+    () => registeredPanels.find((panel) => panel.key === left?.id),
+    [left, registeredPanels]
+  );
+
+  const rightSection = useMemo(
+    () => registeredPanels.find((panel) => panel.key === right?.id),
+    [right, registeredPanels]
+  );
+
+  // retrieve the last preview panel (most recent)
+  const mostRecentPreview = preview ? preview[preview.length - 1] : undefined;
+  const showBackButton = preview && preview.length > 1;
+  const previewSection = useMemo(
+    () => registeredPanels.find((panel) => panel.key === mostRecentPreview?.id),
+    [mostRecentPreview, registeredPanels]
+  );
+
+  const width: number = (leftSection?.width ?? 0) + (rightSection?.width ?? 0);
```

Comment:
Yea, this was a simple solution from my original draft, but I think we should think through defining a percentage based fully expanded flyout and how that could work with the potential components in the panel.

Reply:
let's chat and see if we want to have it as part of this PR or a future one (I don't think I ever got an answer from UX on how it should be done)